lemur/plugins/lemur_azure_dest/tests/test_azure_dest.py

```diff
@@ -122,18 +121,31 @@ def test_upload(self, patched_app):
 
         subject.session.mount("https://", adapter)
 
+        hvac_client_mock().secrets.azure.generate_credentials.return_value = {"client_id": "fakeid123", "client_secret": "fakesecret123"}
+
         name = 'Test_Certificate'
         body = test_server_cert
         private_key = test_server_key
         cert_chain = test_ca_cert
-        options = [{'name': 'vaultUrl', 'value': 'https://couldbeanyvalue.com'}, {'name': 'azureTenant', 'value': 'mockedTenant'},
-                {'name': 'appID', 'value': 'mockedAPPid'}, {'name': 'azurePassword', 'value': 'norealPW'}]
 
-        # commented lines: another unsuccessful try
-        # return value for HTTP post - we won't access Azure
-        # mock_post = Mock()
-        # mock_post.open = mock_open()
-        # mock_post.return_value =  "['message': 'Response', 'status': 200, 'response': {'id': 'someID'}]"
-        plugin.get_access_token = Mock(return_value='valid_test_token')
+        with self.subTest(case="upload cert using azureApp auth method"):
```

Comment:
First time seeing `subTest`, did some reading up on it this morning, very cool 👍